### ISSUE TITLE
Creating github action badges

### ISSUE THREAD

**Issue by muchirijane** [EddieHubCommunity/support#138]
When a beginner starts to contribute they are given a specific badge on their first successful merge. As they continue contributing, they get different badges to encourage them to contribute more🔰 and they also get personalized emails 💌 with their GitHub profile pic.

**Comment by eddiejaoude** [comment]
Love this idea so much 👍 

Could we have a counter that increments each time someone performs an action, then later on we can decide on what badges are for each amount: For example

- `1`: Started badge
- `10`: On a role badge
- ...

**Comment by BOLT04** [comment]
This is super awesomeee, great idea @muchirijane ! @eddiejaoude do you think this Github Action could write some of this data on the database of EddieBot and EddieBotUI? If we have something like this:
```js
{
      username: "Eddie",
      badges: [{
         name: "first-contribution",
         emoji: "🔰", // associated emoji to the badge perhaps??
         description: "This badge is achieved by making a PR and getting a successful merge"
      }]
}
```

then we can display this info on the UI, and show off cool badge UIs 😄.

**Comment by eddiejaoude** [comment]
Yes we could do that in a custom Action with javascript quite easily 👍 

I think we save the statistics against the user, then in another collection or config file, map what it refers to, so we can update that more easily.

For example

Database
```typescript
{
   username: "eddiejaoude"
   // ...
   statistics: {
      issue: {
          created: 10,
          comments: 15
      },
      pullRequest: {
          created: 10,
          comments: 15
       }
   }
}
```

Config
```typescript
{
   badges: {
      pullrequest: {
          1:  {
                name: "first-contribution",
                emoji: "🔰", // associated emoji to the badge perhaps??
                description: "This badge is achieved by making a PR and getting a successful merge"
      } ,
      issues: {
          1:  {
                name: "first-contribution",
                emoji: "🔰", // associated emoji to the badge perhaps??
                description: "This badge is achieved by making an issue"
      } 
   }
}
```

What do you all think? (ok they dont quite align, but something like that, structure needs a bit more thinking)

**Comment by muchirijane** [comment]
That's awesome @eddiejaoude and @BOLT04 . From there, different badges can be created for different levels of successful merges.

**Comment by mikeysan** [comment]
Where is this badge being displaced?

**Comment by eddiejaoude** [comment]
@mikeysan I was thinking we could have a markdown badge available for people to use in their custom GitHub profiles, and also on our web app (I will kick off the `EddieBotUI` next month)

Something like this...

- ![issues](https://img.shields.io/badge/Issues-384-blue?style=flat-square&logo=appveyor)
- ![pull requests](https://img.shields.io/badge/Pull%20Requests-204-blue?style=flat-square&logo=appveyor)

**Comment by eddiejaoude** [comment]
I have started saving the data with a GitHub Action today 🎉 , data from GitHub in Firestore...

```typescript
{
   // GitHub User
   author: {avatar_url: "https://avat...},
   id: "624760",
   issue_comment: 8,
   issues: 4,
   push: 3
}
```

https://github.com/EddieJaoudeCommunity/gh-action-community

**Comment by BOLT04** [comment]
I think this is still a great idea github-actions bot

**Comment by eddiejaoude** [comment]
Yep I agree 👍 . I will remove the label so the bot doesn't close the issue

**Comment by mikeysan** [comment]
Looks like it disobeyed you @eddiejaoude so I have removed the label again. Please revert if that was not your intention.

**Comment by eddiejaoude** [comment]
Thank you @mikeysan 👍

**Comment by eddiejaoude** [comment]
Thanks @schmelto for reopening 👍

**Comment by schmelto** [comment]
Should we transfer this issue to the api / eddiebot?

**Comment by BOLT04** [comment]
@schmelto not sure if this would be implemented in either of those repos. It could be a separate GitHub action like Eddie mentioned [here](https://github.com/EddieHubCommunity/support/issues/138#issuecomment-685132053).

Not sure if a discord bot could implement this, simply because of this requirement: `When a beginner starts to contribute they are given a specific badge on their first successful merge.`. This requires the bot to **know when someone contributed**, so using GitHub WebHooks (where the endpoint would be in the API) is a better use case than the bot. Perhaps there is a way to send these GitHub events through WebSockets, so the bot could react to them but I don't know about it.

How about we define a plan on how to do this and then decide where to transfer the issue? What does everyone think?

**Comment by eddiejaoude** [comment]
I am still keen to do this 👍 it might end up in the LinkFree repo but powered by the API

**Comment by krishguptadev** [comment]
Awesome! Looking forward to it!
